tools/cldr/cldr-to-icu/src/main/java/org/unicode/icu/tool/cldrtoicu/LdmlConverter.java

```diff
@@ -137,76 +138,48 @@ private static PathMatcher supplementalMatcher(String... spec) {
      * hide what are essentially implementation specific data splits.
      */
     public enum OutputType {
-        LOCALES(LDML, LdmlConverter::processLocales),
-        BRKITR(LDML, LdmlConverter::processBrkitr),
-        COLL(LDML, LdmlConverter::processCollation),
-        RBNF(LDML, LdmlConverter::processRbnf),
-
-        DAY_PERIODS(
-            SUPPLEMENTAL,
-            LdmlConverter::processDayPeriods),
-        GENDER_LIST(
-            SUPPLEMENTAL,
-            c -> c.processSupplemental("genderList", GENDER_LIST_PATHS, "misc", false)),
-        LIKELY_SUBTAGS(
-            SUPPLEMENTAL,
-            c -> c.processSupplemental("likelySubtags", LIKELY_SUBTAGS_PATHS, "misc", false)),
-        SUPPLEMENTAL_DATA(
-            SUPPLEMENTAL,
-            c -> c.processSupplemental("supplementalData", SUPPLEMENTAL_DATA_PATHS, "misc", true)),
-        CURRENCY_DATA(
-            SUPPLEMENTAL,
-            c -> c.processSupplemental("supplementalData", CURRENCY_DATA_PATHS, "curr", true)),
-        METADATA(
-            SUPPLEMENTAL,
-            c -> c.processSupplemental("metadata", METADATA_PATHS, "misc", false)),
-        META_ZONES(
-            SUPPLEMENTAL,
-            c -> c.processSupplemental("metaZones", METAZONE_PATHS, "misc", false)),
-        NUMBERING_SYSTEMS(
-            SUPPLEMENTAL,
-            c -> c.processSupplemental("numberingSystems", NUMBERING_SYSTEMS_PATHS, "misc", false)),
-        PLURALS(
-            SUPPLEMENTAL,
-            LdmlConverter::processPlurals),
-        PLURAL_RANGES(
-            SUPPLEMENTAL,
-            LdmlConverter::processPluralRanges),
-        WINDOWS_ZONES(
-            SUPPLEMENTAL,
-            c -> c.processSupplemental("windowsZones", WINDOWS_ZONES_PATHS, "misc", false)),
-        TRANSFORMS(
-            SUPPLEMENTAL,
-            LdmlConverter::processTransforms),
-        KEY_TYPE_DATA(
-            BCP47,
-            LdmlConverter::processKeyTypeData),
-
-        // Batching by type.
-        DTD_LDML(LDML, c -> c.processAll(LDML)),
-        DTD_SUPPLEMENTAL(SUPPLEMENTAL, c -> c.processAll(SUPPLEMENTAL)),
-        DTD_BCP47(BCP47, c -> c.processAll(BCP47));
-
-        public static final ImmutableSet<OutputType> ALL =
-            ImmutableSet.of(DTD_BCP47, DTD_SUPPLEMENTAL, DTD_LDML);
+        LOCALES(LDML),
```

Comment:
Simplify the enum to stop it doing dispatching (that's now done below).
This is needed because the filtering work wants to operate on a single data instance (previously we would run over the same data multiple times, depending on the output types). Now we just map the data from a CLDR supplier once and then split it afterwards.